[mlir][vector][nfc] Update vector-to-llvm.mlir

banach-space opened the pull request:
* Adds extra comments to group Ops
* Unifies the test function naming, i.e.
  * `@vector_{op_name}_{variant}` -> `@{op_name}_{variant}`
* Unifies input variable names (`%input` -> `%arg0`)
* Capitalises LIT variable names (e.g. `%[[insert]]` --> `%[[INSERT]]`)
* Moves `@step_scalable()` _below_ its "fixed-width" counterpart
  (to follow the existing consistency within this file).

There's still some inconsistencies within this file - I'm happy to send
more updates if folks find it useful. But I'd definitely recommend
splitting across multiple PRs (otherwise it's hard to review).

banach-space committed:
[mlir][vector][nfc] Update vector-to-llvm.mlir

* Adds extra comments to group Ops
* Unifies the test function naming, i.e.
  * `@vector_{op_name}_{variant}` -> `@{op_name}_{variant}`
* Unifies input variable names (`%input` -> `%arg0`)

There's still some inconsistencies within this file - I'm happy to send
more updates if folks find it useful. But I'd definitely recommend
splitting across multiple PRs (otherwise it's hard to review).

banach-space commented:
Gentle ping @nujaa :) This includes arg re-names that you suggested recently :)

nujaa commented on mlir/test/Conversion/VectorToLLVM/vector-to-llvm.mlir:
```diff
@@ -1,79 +1,81 @@
 
-// TODO: Add tests for for vector.type_cast that would cover scalable vectors
+//===----------------------------------------------------------------------===//
+// vector.bticast
```
```suggestion
// vector.bitcast
```

nujaa commented on mlir/test/Conversion/VectorToLLVM/vector-to-llvm.mlir:
```diff
@@ -2900,12 +2980,29 @@ func.func @flat_transpose_index(%arg0: vector<16xindex>) -> vector<16xindex> {
 
-func.func @vector_load(%memref : memref<200x100xf32>, %i : index, %j : index) -> vector<8xf32> {
+//===----------------------------------------------------------------------===//
+// vector.load
```
This comment is misplaced and should be after flat_transpose tests.

nujaa commented on mlir/test/Conversion/VectorToLLVM/vector-to-llvm.mlir:
```diff
@@ -3696,42 +3844,46 @@ func.func @vector_deinterleave_2d_scalable(%a: vector<2x[8]xf32>) -> (vector<2x[
 // CHECK: %[[CAST:.*]] = builtin.unrealized_conversion_cast %[[STEPVECTOR]] : vector<[4]xi64> to vector<[4]xindex>
 // CHECK: return %[[CAST]] : vector<[4]xindex>
-func.func @vector_step_scalable() -> vector<[4]xindex> {
+func.func @step_scalable() -> vector<[4]xindex> {
```
NIT or part of a future consistency patch: The scalable case is usually after the standard one.

banach-space replied:
Let me do it here - no need to artificially create PR traffic.

nujaa reviewed (commented):
Hi, thanks for your continuous effort. Here are a few nits :)
In terms of remaining inconsistencies, I found 
* inputs named %a, %b, ... (I am ok to have both the (mem, vec, i) naming and argX in the same file)
* There are some missing delimiter `// -----`
Did you think of something else ?

banach-space committed:
fixup! [mlir][vector][nfc] Update vector-to-llvm.mlir

* Add missing "split" seperators (`// -----`)
* Capitialize LIT variables (e.g. %[[insert]] --> %[[INSERT]])
* Renamed some func variables to either better align with the convention in the
  file (e.g. `%a` -> `%vec_1d`) or to make sure the name is more
  descriptive (e.g. `%a` -> `%num_elems`)
* Other typos highlighted by Hugo (thanks!)

banach-space commented:
> In terms of remaining inconsistencies, I found
> 
> * inputs named %a, %b, ... (I am ok to have both the (mem, vec, i) naming and argX in the same file)
> * There are some missing delimiter `// -----`

These should be now fixed. 

>   Did you think of something else ?

I think that tests for `vector.fma` should be split into 3 cases:
* 0D vectors,
* 1D vectors,
* 2D vectors.

But I see that as a low priority. Splitting these tests into multiple files would also be a good idea.

nujaa commented on mlir/test/Conversion/VectorToLLVM/vector-to-llvm.mlir:
```diff
@@ -2980,23 +3144,27 @@ func.func @flat_transpose_index(%arg0: vector<16xindex>) -> vector<16xindex> {
+// -----
+
+//===----------------------------------------------------------------------===//
+// vector.load
```
Ayy, this should be moved after flat_transpose LIT lines.

banach-space replied:
Argh, sorry about that, looked OK on my screen  🤦🏻 Thanks!

nujaa commented on mlir/test/Conversion/VectorToLLVM/vector-to-llvm.mlir:
```diff
@@ -3802,6 +3970,8 @@ func.func @deinterleave_1d(%a: vector<4xi32>) -> (vector<2xi32>, vector<2xi32>)
+
 // CHECK-LABEL: @deinterleave_1d_scalable
 // CHECK-SAME:  %[[SRC:.*]]: vector<[4]xi32>) -> (vector<[2]xi32>, vector<[2]xi32>)
 func.func @deinterleave_1d_scalable(%a: vector<[4]xi32>) -> (vector<[2]xi32>, vector<[2]xi32>) {
```
Forgotten %a, %b in the deinterleave tests.

nujaa reviewed (commented):
Hello, a few more changes and we'll  reach perfection. 
I also noted a discrepancy between LIT variable settings. Some use [[VEC:[0-9a-zA-Z]+]] or [[VEC:.*]] . But I don't personally care.

banach-space commented:
>  Some use [[VEC:[0-9a-zA-Z]+]] or [[VEC:.*]] . 

This one is actually a bit tricky to fix - it requires a few more changes in the captured output. And this PR is getting a bit noise. Shall we keep this for the future?

banach-space committed:
fixup! fixup! [mlir][vector][nfc] Update vector-to-llvm.mlir

* Further capitalization fixes
* Moved `vector.load` comment
* Rename func variables (`%a` -> `%arg`)

banach-space commented:
>   Did you think of something else ?

`vector.reduction` tests are split (and most likely duplicated) across two files:
* https://github.com/llvm/llvm-project/blob/main/mlir/test/Conversion/VectorToLLVM/vector-reduction-to-llvm.mlir
* https://github.com/llvm/llvm-project/blob/main/mlir/test/Conversion/VectorToLLVM/vector-to-llvm.mlir

nujaa reviewed (approved):
LGTM, thanks. This MR became way larger than originally meant. You could eventually update the commit summary to mention the post review updates. e,g. %a -> %arg, LIT variables uppercase-ation
